[CS2113-T13-2] Domnus

johanesrafael opened the pull request:
Domnus is a portable timetable scheduler for NUS students and provides 3 main feature namely: Task Tracking, Timetabling and Cap Calculating features. This application aims to increase the productivity of students in the context of having an effective task tracking software product.

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
 # Developer Guide
 
-## Design & implementation
+## 1. Table of content
```
Consider using links to the different sections for the table of contents? It will ensure easier navigability of the document.

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+6.  Type ‘bye’ to terminate your session.
 
-## Product scope
+## 3. Design
```
Consider adding 1-2 lines preface for this section, to introduce the points covered here.

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+### 3.1 Architecture
+The **Architecture Diagram** below represents a high-level design overview of the App. Specifically, it is done with an **N-tier architectural style**, where the higher layers make use of services provided by lower layers. 
+
```
Nicely described system architecture!

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+Main gets user input and displays messages through the use of UI component. 
+The UI layer entails the package *visualize*, which contains classes *ColoredString*, *Bitmap*, *UI*, *Cli*, 
+*FancyCli* and enumerations *Color* and *Sprite* in the following structure:
+![uml](Images/Package%20visualize.png)
```
Perhaps use the standard notation for class diagrams? Instead of icons for class, interface, enum etc. it would be good if it follows the textbook notation. Eg. Instead of the icon E in shown for the enumerations, using <<enumeration>> instead..

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+**1.6 Flow of DOMSUN**<br>
+The sequence diagram below shows the main interaction of classes with each other throughout the whole lifecycle of DOMSUM.
+![uml](Images/DOMSUM_Main_Flow.png)
```
Some thing about the sequence diagram:
* The activation bars begin before the call arrows and end after the return arrows, which is an incorrect notation. Consider changing the activation bars to begin at the arrow call and end at the return call?
* The activation bar for `:DOMNUS` in this diagram ends within the loop itself, thus implying the program execution terminates within the loop. I'd suggest extending the activation bar a bit more?

Otherwise a nicely done diagram!

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+1. blah
+
+### 4.2 Checker Feature 
+![here](Images/Checker_Diagram.png)
```
I think some the class names are missing the ":" in front of them. Not sure if this is an optional/necessary convention but perhaps consider adding the colons?

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![cap uml diagram](Images/CalculateCapSequence.png)
```
I think its a glitch of some sort, but the background of the image seems to be transparent so it is appearing black so I can't seem much in the diagram. 
![image](https://user-images.githubusercontent.com/57522870/97396179-84f20200-1921-11eb-9da3-a2e7fa7f86d0.png)

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following sequence diagram diagram shows how the reminder operation works
+
+![Reminder_Sequence_Diagram](Images/ReminderAction_Sequence_Diagram.png)
```
Same issue as the previous diagram :(

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Remind.png)
```
The first call arrow in this sequence diagram starts at the middle of the activation bar, and there is no return arrow from the function call to represent transfer of control back to the caller.

![image](https://user-images.githubusercontent.com/57522870/97396640-6dffdf80-1922-11eb-8b74-e25efa64aa2a.png)

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
-## User Stories
-
-|Version| As a ... | I want to ... | So that I can ...|
+|Priority| As a ... | I want to ... | So that I can ...|
```
A nice and comprehensive list of user stories!

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+ - Mainstream OS: Windows, Linux, Unix, OS-X
+ - Private contact detail 
+
+## Appendix F. Instructions for manual testing
```
This part is well written! :D

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/PostponeAction_Sequence_Diagram.png)
```
A fairly well done sequence diagram! However, I didn't quite get what was happening in this part:
![image](https://user-images.githubusercontent.com/57522870/97397196-ae139200-1923-11eb-9a6d-7ef6ecef5d7a.png)
The control is returned not the calling class, but the other class? Consider doing it like this maybe?: 
![image](https://user-images.githubusercontent.com/57522870/97397414-1ebaae80-1924-11eb-9c70-4a837152e1d7.png)

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Grade_Sequence_Diagram](Images/GradeSequence.png)
```
Similar transparent background issue, perhaps it might've occurred whilst rendering the image?

sevenseasofbri commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,407 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Focus.png)
```
I think a similar issue with the activation bars starting before the arrow here?

sevenseasofbri reviewed (commented):
### Overall, a very well done Dev Guide! 💯 
Just a few suggestions:
* Links to the ToC leading to each section.
* Figure numbers.
One thing I noticed was that the sequence diagrams were probably done using different softwares as they're all sort of different from each other. Perhaps standardize the diagrams?

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+## 2. Setting Up
 
+1.  Ensure that you have Java 11 or above installed.
+2.  Download the latest version of  `Duke`  from  [Our Release Page](https://github.com/AY2021S1-CS2113-T13-2/tp/releases/tag/v1.0).
```
Do remember to update the link to the release page

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+**1.6 Flow of DOMSUN**<br>
+The sequence diagram below shows the main interaction of classes with each other throughout the whole lifecycle of DOMSUM.
+![uml](Images/DOMSUM_Main_Flow.png)
```
The layout of the image looks wrong on the website. Could it be solved with a whitespace between the text and the image?

Wu-Haitao replied:
This diagram have some indentation before it, which is not consistent with the other ones. And I found it's because there is a line "lifecycle of DOMSUM." before the diagram. So please try separate them.
![image](https://user-images.githubusercontent.com/59434361/97666908-99690280-1ab9-11eb-8c0e-062263d28e79.png)

Johnson-Yee replied:
Hey there, some comments about the sequence diagram here.
1. The call arrows should begin at the top of the activation bar. The return arrows should be at the bottom fo the activation bar.
2. I noticed that there is also an alt block for isFancy, not sure of its use there as there is no activation block in it or no method called.
3. After the return arrow of the savefile, is there a reason for the extension of activation bar for :DomNUS?

Johnson-Yee replied:
In addition to the points made previously, the description of some arrows could be changed eg there are two arrows stating "Parser", not sure of its meaning, isit calling a function?
Otherwise, the diagram looks good!

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+## 4. Implementation<br>
+This section highlights some of our project's key feature and its implementation. 
+
+### 4.1 Take Feature
```
Would it be more easy to explain with a sequence diagram?

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+Step 5. Now we proceed to call `checkDuplicates()` of Checker class. 
+
+Step 6. If `false` , there is no duplicates in the existing list, and the task can be safely added. Otherwise, no action will be taken. 
```
Very clear explanation!

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Snooze.png)
```
Activation bar of method getNewInterval() starts before the method call arrives in this diagram

WYing333 commented on docs/DeveloperGuide.md:
```diff
@@ -132,14 +132,13 @@ Step 4. CAP value is calculated and returned to the user through `Ui`.
 
-![cap uml diagram](Images/Cap_Calculator_Diagram.JPG)
-
+![cap uml diagram](Images/CalculateCapSequence.png)
```
Cannot see the lines and descriptions of the diagrams clearly. Maybe can change the color of the background or the lines.
<img width="785" alt="diagram" src="https://user-images.githubusercontent.com/69497423/97544273-51d46f00-1a04-11eb-8923-520605aa9c38.png">

WYing333 commented on docs/DeveloperGuide.md:
```diff
@@ -177,7 +176,28 @@ The following activity diagram summarizes what happens when a user executes a ne
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Grade_Sequence_Diagram](Images/GradeSequence.png)
```
Same issue as the one described above.
<img width="1070" alt="diagramm" src="https://user-images.githubusercontent.com/69497423/97545546-25215700-1a06-11eb-93b7-612429b0be44.png">

WYing333 commented on docs/DeveloperGuide.md:
```diff
@@ -214,7 +272,7 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 ![Postpone_Sequence_Diagram](Images/PostponeAction_Sequence_Diagram.png)
```
Overall, the diagrams are quite clear. However, maybe the way to draw diagrams could be consistent.
<img width="680" alt="3" src="https://user-images.githubusercontent.com/69497423/97545461-04f19800-1a06-11eb-9862-e1dd41cc67a4.png">
<img width="1030" alt="4" src="https://user-images.githubusercontent.com/69497423/97545471-07ec8880-1a06-11eb-8ac6-9625ea2d87e4.png">

WYing333 commented on docs/DeveloperGuide.md:
```diff
@@ -192,7 +250,7 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 ![Postpone_Sequence_Diagram](Images/Snooze.png)
```
The arrow of the method getNewInterval() needs to be at the start of the activation bar.
<img width="647" alt="bar" src="https://user-images.githubusercontent.com/69497423/97546287-30c14d80-1a07-11eb-8347-9095f495d795.png">

Wu-Haitao commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![cap uml diagram](Images/CalculateCapSequence.png)
```
This diagram, as well as the next one for 4.5 Reminder Feature, are all have a black(or transparent) background, which makes it quite hard to follow  the lines and arrows. Please check if you exported the UML in a proper way and try to solve it.
![image](https://user-images.githubusercontent.com/59434361/97667224-63784e00-1aba-11eb-8102-f7731c48117d.png)

Johnson-Yee replied:
Same problem encountered as the above reviewers causing the diagram to be messy. Maybe there was a problem with the picture file.

Johnson-Yee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
 # Developer Guide
 
-## Design & implementation
+## 1. Table of content
```
Slight error here: Table of contents instead of content.
Do make sure the links work as well!

Wu-Haitao commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| *** |Student before start of semester|List the modules MC|Follow the recommended MC |
+| *** |Student before start of semester|List of module available|Easily Choose which modules to take|
+| ** |Student before start of semester|Find the modules either by keyword, module code or even MC |Easily see the desired modules |
+| ** |Student before start of semester|Select the modules but not taking it yet |Easily whether the MC fits my requirement |
+| *  |Student before start of semester|Find out the etails of the Module|To find out more about the modules.|
+| *** |Student before start of semester|Take the desired modules|Mark the modules that i want to take as taken  |
+| *** |Student during the semester|Add tasks such as todo,deadline and event into my list|Easily keep track of all the task i have to complete  |
+| ** |Student during the semester|Have a Reminder of which deadline is due soon|Ensure that no task is missed out  |
+| *** |Student during the semester|Add task to modules |Easily know which tasks belongs to which modules  |
+| *** |Student during the semester|Delete task once they are completed |Remove unnecessary task on the list |
+| ** |Student after the semester|Calculate the CAP of my individual modules|Easily find out my performance this semester |
+| * |Student after the semester|Clear the list of tasks and modules|Start afresh for the next semester |
```
I can see that currently you split the user stories into three groups which is good. But as you have already prioritized them, I suggest that you could reorder them by their priorities.
For example:
*** before
**  before
*** during
** during
...

Johnson-Yee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+Given below is an example usage scenario and how the checker mechanism behaves at each step. 
+
+![here](Images/Checker_Diagram.png)
```
Could include activation bars in your sequence diagram.
Otherwise, good job!
![image](https://user-images.githubusercontent.com/60440731/97668261-8d327480-1abc-11eb-9993-2743a6e056c4.png)

Wu-Haitao commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+*FancyCli* and enumerations *Color* and *Sprite* in the following structure:
+![uml](Images/Package%20visualize.png)
+
+UI's interaction with user<br>
+UI gets user input through *nextline()*, and renders strings as a user-comprehensible interface through 
+*update(String, Data)*.
+
+UI's interaction with the rest of the program<br>
+UI passes the user's input string out to the Duke object, which then passes the string to the Command Interpreter layer.
+UI also reads data from the Data object for refreshing purposes, but does not modify it. 
```
Maybe it's better to rearrange this part. It looks a little bit messy.
And for these two *nextline()* and *update(String, Data)* do you mean `nextline()` and `update(String, Data)` instead?

Wu-Haitao commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+Step 7. Now `StatsAction` is completed and it will return this string back to `Execute` for to be printed through `UI`. 
+
+
+**Design consideration:**
```
You could delete this line if there is no content.

Wu-Haitao commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,513 @@
+
+Step 6. If `false` , there is no duplicates in the existing list, and the task can be safely added. Otherwise, no action will be taken. 
+
+**Design consideration: 
+Aspect: How checker executes**
```
This part looks problematic on the page. I guess you forgot to add "***" at the right places?
![image](https://user-images.githubusercontent.com/59434361/97668780-af78c200-1abd-11eb-928e-253d11cc1d82.png)

Wu-Haitao reviewed (commented):
The DG is generally good. 👍

adinata15 committed:
Merge pull request #257 from adinata15/v2.1_documentation

Fix table in GitHub UserGuide page

scjx123 committed:
Update UserGuide.md

chuhann committed:
Restructure userguide flow

chuhann committed:
Merge pull request #265 from chuhann/V2.1_UserGuide

Restructure userguide flow

chuhann committed:
Update userguide with edit functionality

chuhann committed:
Merge pull request #266 from chuhann/V2.1_UserGuide

Update userguide with edit functionality

scjx123 committed:
Update DeveloperGuide.md

scjx123 committed:
Update DeveloperGuide.md

adinata15 committed:
Merge pull request #38 from AY2021S1-CS2113-T13-2/master

Update upstraeam

chuhann committed:
Reorganize userguide

chuhann committed:
Merge pull request #267 from chuhann/V2.1_UserGuide

Reorganize userguide

scjx123 committed:
Merge pull request #20 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Add Checker Exception

scjx123 committed:
Edit Checker Diagram

scjx123 committed:
Sync repo

scjx123 committed:
edit checkstyle

scjx123 committed:
Refactor Domnus into Domsun

scjx123 committed:
Merge pull request #268 from scjx123/master

Enable todo for Checker

scjx123 committed:
Update DeveloperGuide.md

scjx123 committed:
Merge pull request #21 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Add UG diagram

scjx123 committed:
Update UserGuide.md

scjx123 committed:
Merge pull request #269 from scjx123/master

Update UG

scjx123 committed:
Update README.md

chuhann committed:
Update UserGuide.md

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes

TomLBZ committed:
Closes #263

Closes #263

TomLBZ committed:
Closes #263 : fix tests

Closes #263 : fix tests

TomLBZ committed:
Closes #262.

Closes #262.

TomLBZ committed:
Closes 262 : fix format

Closes 262

TomLBZ committed:
Closes #262 : fix format

Closes #262

TomLBZ committed:
Closes #262 : fix tests

Closes #262

TomLBZ committed:
Merge remote-tracking branch 'origin/v2.1_bugFixes' into v2.1_bugFixes

TomLBZ committed:
UI update: Added word wrapping and help prompt. Changed banner behaviour

TomLBZ committed:
Bug fix: Fixed the unintended memory effect for Next and Prev Commands

TomLBZ committed:
Closes #260

Closes #260

TomLBZ committed:
Closes #260 : fix format

Closes #260

TomLBZ committed:
Closes #258

Closes #258

TomLBZ committed:
Closes #245

Closes #245

TomLBZ committed:
Closes #243

Closes #243

TomLBZ committed:
Closes #251

Closes #251

TomLBZ committed:
Closes #241

Closes #241

TomLBZ committed:
Closes #239

Closes #239
Also, UI update: improved wrapping to be word wrapping instead of letter wrapping in the bottom region.

TomLBZ committed:
Closes #238

Closes #238
Also improved clear action.

TomLBZ committed:
Closes #237

Closes #237

TomLBZ committed:
Closes #232

Closes #232

chuhann committed:
Update checker diagram

chuhann committed:
Merge pull request #271 from chuhann/master

Update checker diagram

scjx123 committed:
Merge pull request #22 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Add StatsAction diagram

TomLBZ committed:
Closes #228

Closes #228

TomLBZ committed:
Merge pull request #270 from TomLBZ/v2.1_bugFixes

V2.1 bug fixes, fixed 13 bugs, re-implemented 3 classes, updated 2 classes

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master'

TomLBZ committed:
Closes #221, Closes #259, Closes #244

Closes #221, Closes #259, Closes #244

chuhann committed:
Update broken links for userguide and reformat userguide structure

chuhann committed:
Update command summary table format

scjx123 committed:
add PPP template

TomLBZ committed:
Closes #264

Closes #264

scjx123 committed:
Correct typo in PPP

scjx123 committed:
Merge pull request #23 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Update DeveloperGuide.md

scjx123 committed:
Update DeveloperGuide.md

TomLBZ committed:
ppp draft, edit action base structure

scjx123 committed:
Merge pull request #274 from scjx123/master

Add PPP

scjx123 committed:
Add notes field

chuhann committed:
Merge pull request #273 from chuhann/V2.1_UserGuide

V2.1 user guide

scjx123 committed:
Edit StatsAction Test and Re-add Weekly Recurrence

scjx123 committed:
Merge branch 'master' of https://github.com/scjx123/tp

scjx123 committed:
Edit checkstyle

scjx123 committed:
Merge pull request #275 from scjx123/master

add notes field

scjx123 committed:
Update UserGuide.md

scjx123 committed:
Update DeveloperGuide.md

adinata15 committed:
Merge pull request #39 from AY2021S1-CS2113-T13-2/master

Update upstream

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes2

chuhann committed:
Restructure user guide flow

chuhann committed:
Merge pull request #277 from chuhann/V2.1_UserGuide

Restructure user guide flow

johanesrafael committed:
Reformat

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_Documentation

# Conflicts:
#	src/main/java/seedu/duke/Domsun.java

johanesrafael committed:
Remove unpeer-reviewed function

adinata15 committed:
Edit Helptext and Minor test

adinata15 committed:
Merge pull request #290 from adinata15/master

Edit helptext and minor testing

scjx123 committed:
Update scjx123.md

scjx123 committed:
Merge pull request #24 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Add comments

scjx123 committed:
update StatsActionTest

scjx123 committed:
Merge pull request #291 from scjx123/master

update StatsActionTest

adinata15 committed:
Merge pull request #40 from AY2021S1-CS2113-T13-2/master

update upstream

chuhann committed:
Fix minor formatting issue for userguide

chuhann committed:
Merge pull request #292 from chuhann/V2.1_Userguide

Fix minor formatting issue for userguide

adinata15 committed:
Add more testing cases

adinata15 committed:
Merge pull request #293 from adinata15/v2.1_bug_edit

Edit and add test cases

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_Documentation

johanesrafael committed:
Merge pull request #295 from johanesrafael/V2.1_FixBug

Remove unpeer-reviewed function

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes2

adinata15 committed:
Merge pull request #41 from AY2021S1-CS2113-T13-2/master

update upstream

chuhann committed:
Update userguide formatting

chuhann committed:
Update help text section for postpone and reminder

johanesrafael committed:
Fix Invalid Date for Event or Deadline Bug

johanesrafael committed:
Reformat constant

chuhann committed:
Merge pull request #296 from chuhann/V2.1_Userguide

V2.1 userguide

johanesrafael committed:
Merge pull request #297 from johanesrafael/V2.1_FixBug

Fix Postponing invalid Date for event or deadline Bug

adinata15 committed:
Add ModuleAction Test

adinata15 committed:
Edit PPP and refactor code

adinata15 committed:
Merge pull request #298 from adinata15/v2.1_bug_edit

Finalize normal test case

TomLBZ committed:
Closes #289

Closes #289

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes2

TomLBZ committed:
Merge With Formatting

TomLBZ committed:
Closes #288

Closes #288

TomLBZ committed:
Closes #287

Closes #287

johanesrafael committed:
Edit PPP

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_FixBug

TomLBZ committed:
Closes #281

Closes #281

johanesrafael committed:
Add lttle detail in UG

TomLBZ committed:
Closes #280

Closes #280

scjx123 committed:
Update UserGuide.md

TomLBZ committed:
Closes #279

Closes #279

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes2

scjx123 committed:
Update UserGuide.md

scjx123 committed:
Update UserGuide.md

TomLBZ committed:
Base commit

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_bugFixes2

TomLBZ committed:
Merge branch 'v2.1_bugFixes2' into v2.1_enhancements

TomLBZ committed:
Merge pull request #272 from TomLBZ/v2.1_bugFixes2

V2.1 bug fixes2

johanesrafael committed:
Reformat SnoozeAction

johanesrafael committed:
Update HelpText

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_FixBug

# Conflicts:
#	src/main/java/constants/HelpText.java

johanesrafael committed:
Update helptext indentation.

johanesrafael committed:
Merge pull request #299 from johanesrafael/V2.1_FixBug

Reformat SnoozeAction and Update HelpText

TomLBZ committed:
Closes #224

Closes #224

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_enhancements

TomLBZ committed:
Merge branch 'v2.1_bugFixes2'

TomLBZ committed:
Closes #222

Closes #222

TomLBZ committed:
Closes #222 : fix formats

Closes #222 : fix formats

TomLBZ committed:
Bug fixes on Edit action

Bug fixes on Edit action

TomLBZ committed:
Closes #240

Closes #240

TomLBZ committed:
Merge pull request #300 from TomLBZ/v2.1_enhancements

V2.1 enhancements

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master'

adinata15 committed:
Merge pull request #42 from AY2021S1-CS2113-T13-2/master

update upstream

adinata15 committed:
Minor edit on Helptext

adinata15 committed:
Add more test.
Implement LoadActionTest and modify PostponeActionTest.

adinata15 committed:
Refactor code

adinata15 committed:
Merge pull request #301 from adinata15/v2.1_bug_edit

Implement additional test

johanesrafael committed:
update HelpText

johanesrafael committed:
Update UserGuide

johanesrafael committed:
Add authorship

johanesrafael committed:
Update ReminderActionTest

johanesrafael committed:
Add SnoozeActionTest

johanesrafael committed:
Implement PostponeActionTest

johanesrafael committed:
Add test_task

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_FixBug

# Conflicts:
#	src/test/java/command/action/PostponeActionTest.java

johanesrafael committed:
reformat

johanesrafael committed:
Update PostponeActionTest

johanesrafael committed:
Update PPP

johanesrafael committed:
Merge pull request #302 from johanesrafael/V2.1_FixBug

Update UG, HelpText, SnoozeActionTest,  ReminderActionTest, PostponeActionTest, and start filling PPP

johanesrafael committed:
update README.md

scjx123 committed:
Merge pull request #25 from AY2021S1-CS2113-T13-2/master

sync repo

scjx123 committed:
Update UserGuide.md

scjx123 committed:
Update DeveloperGuide.md

chuhann committed:
Update sequence diagram image

chuhann committed:
Update Userguide format

chuhann committed:
Update Aboutus image

chuhann committed:
Add PPP

scjx123 committed:
Add diagram

scjx123 committed:
Merge branch 'master' of https://github.com/scjx123/tp

scjx123 committed:
Update DeveloperGuide.md

chuhann committed:
Merge pull request #304 from chuhann/V2.1_UserGuide

V2.1 user guide

adinata15 committed:
Merge pull request #43 from AY2021S1-CS2113-T13-2/master

update upstream

scjx123 committed:
Fix Checkstyle

scjx123 committed:
Merge branch 'master' of https://github.com/scjx123/tp

scjx123 committed:
Update DeveloperGuide.md

scjx123 committed:
Merge pull request #306 from scjx123/master

Fix grammar DG, UG

chuhann committed:
Remove ``` from user guide

chuhann committed:
Merge pull request #307 from chuhann/V2.1_UserGuide

Remove ``` from user guide

chuhann committed:
Remove more ``` for formatting issue

chuhann committed:
Merge pull request #308 from chuhann/V2.1_UserGuide

Remove more ``` for formatting issue

adinata15 committed:
Merge pull request #44 from AY2021S1-CS2113-T13-2/master

update upstream

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_FixBug

adinata15 committed:
Finish Helptext notes

adinata15 committed:
Refactor code

adinata15 committed:
Merge pull request #309 from adinata15/v2.1_documentation

Add Helptext notes

scjx123 committed:
Fix MC bug

scjx123 committed:
Merge branch 'master' of https://github.com/scjx123/tp

scjx123 committed:
Merge pull request #310 from scjx123/master

Fix MC bug

adinata15 committed:
Merge pull request #45 from AY2021S1-CS2113-T13-2/master

update upstream

adinata15 committed:
Minor edit to Helptext

adinata15 committed:
Merge branch 'master' of https://github.com/adinata15/tp into master

chuhann committed:
Remove reminder feature from developer guide

chuhann committed:
Merge pull request #311 from chuhann/master

Remove reminder feature from developer guide

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_preRelease

adinata15 committed:
Add symbols explanation on UserGuide
Include section to explain use/meaning of different symbol on UserGuide.

adinata15 committed:
Merge pull request #312 from adinata15/v2.1_documentation

Add symbol explanation on UserGuide

adinata15 committed:
Merge pull request #46 from AY2021S1-CS2113-T13-2/master

update upstream

chuhann committed:
Update user guide with load functionality

chuhann committed:
Merge pull request #313 from chuhann/master

Update user guide with load functionality

adinata15 committed:
Refactor code

adinata15 committed:
Fix checktyle

adinata15 committed:
Merge pull request #314 from adinata15/master

Refactor code

TomLBZ committed:
Update PPP (unfinished)

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_preRelease

chuhann committed:
Update invidiual PPP

chuhann committed:
Merge pull request #315 from chuhann/master

Update invidiual PPP

johanesrafael committed:
Update AboutUs.md

chuhann committed:
Update PPP

chuhann committed:
Fix formatting issue for PPP

chuhann committed:
Fix formatting for PPP

chuhann committed:
Minor line break fix to PPP

chuhann committed:
Change line break to sub points for PPP

chuhann committed:
Add PPP

chuhann committed:
Update PPP

adinata15 committed:
Merge pull request #47 from AY2021S1-CS2113-T13-2/master

update upstream

chuhann committed:
Merge pull request #316 from chuhann/master

Add PPP

scjx123 committed:
Update scjx123.md

scjx123 committed:
Update scjx123.md

scjx123 committed:
Update DeveloperGuide.md

chuhann committed:
no message

scjx123 committed:
Update DeveloperGuide.md

scjx123 committed:
Merge pull request #26 from AY2021S1-CS2113-T13-2/master

sync repo

TomLBZ committed:
Updated UG, PPP

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_preRelease

scjx123 committed:
Update DeveloperGuide.md

TomLBZ committed:
Merge pull request #317 from TomLBZ/v2.1_preRelease

V2.1 pre release

johanesrafael committed:
Update PPP

johanesrafael committed:
Merge remote-tracking branch 'upstream/master' into V2.1_FixBug

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master'

johanesrafael committed:
Update PPP

johanesrafael committed:
Format PPP

scjx123 committed:
Update UserGuide.md

scjx123 committed:
Update UserGuide.md

johanesrafael committed:
Merge pull request #318 from johanesrafael/V2.1_FixBug

update PPP

chuhann committed:
Update user guide with expected output

chuhann committed:
Update todo

chuhann committed:
Merge pull request #319 from chuhann/master

Update userguide

johanesrafael committed:
Edit PPP

scjx123 committed:
Update UserGuide.md

adinata15 committed:
Edit DG Sequence Diagram

johanesrafael committed:
Merge pull request #320 from johanesrafael/V2.1_FixBug

Edit PPP Again!!

TomLBZ committed:
update docs

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master' into v2.1_preRelease

TomLBZ committed:
Merge pull request #322 from TomLBZ/v2.1_preRelease

V2.1 pre release

adinata15 committed:
Merge pull request #48 from AY2021S1-CS2113-T13-2/master

update upsteam

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master'

scjx123 committed:
Update UserGuide.md

adinata15 committed:
Fix style

adinata15 committed:
Merge pull request #323 from adinata15/master

Update sequence diagram

scjx123 committed:
Merge branch 'master' into master

TomLBZ committed:
edit PPP

TomLBZ committed:
Merge remote-tracking branch 'tp-parent/master'

TomLBZ committed:
Merge pull request #324 from TomLBZ/master

update ppp

TomLBZ committed:
final ppp

TomLBZ committed:
Merge pull request #326 from TomLBZ/master

final ppp
